zerver/lib/url_encoding.py

```diff
@@ -74,15 +89,17 @@ def near_stream_message_url(realm: Realm, message: Dict[str, Any]) -> str:
         "topic",
         encoded_topic,
         "near",
-        message_id,
+        message_id_,
     ]
     full_url = "/".join(parts)
     return full_url
 
 
-def near_pm_message_url(realm: Realm, message: Dict[str, Any]) -> str:
-    message_id = str(message["id"])
-    str_user_ids = [str(recipient["id"]) for recipient in message["display_recipient"]]
+def near_pm_message_url(
+    realm: Realm, message_id: int, display_recipient: List[UserDisplayRecipient]
+) -> str:
+    msg_id = str(message_id)
+    str_user_ids = [str(recipient["id"]) for recipient in display_recipient]
```

Comment:
You could probably call this `user_id_strings = [str(recipient_id["id"] for recipient_user in ...]` for similar reasons to the above